docs/oauth-setup.md

```diff
@@ -0,0 +1,143 @@
+    - <github_tenant_name> - this can be any name you would like.  SHIELD will create an internal SHIELD tenant with this given name
+    - rights - this section will contain the mappings from Github team to a predefined SHIELD role, as wel as a default role for all users that don't belong to any of the specified Github teams
+      - the shield roles are as follows:
+        - admin
+        - operator
+        - engineer
+
+#### Here is an example config used for SHIELD development:
+
+```yaml
+auth:
+  - name:       Github
+    identifier: public-gh
+    backend:    github
+    properties:
+      client_id:     435abbcea49c948f8ab3
+      client_secret: ffab373bc287d40ba73476bff3f449
+      mapping:
+        starkandwayne:           # <-- github org name
+          tenant: starkandwayne  # <-- shield tenant name
+          rights:
+            - team: Owners       # <-- github team name
+              role: admin        # <-- shield role name
+            - team: Engineering  #   (first match wins)
+              role: engineer
+            - role: operator     # = (default match)
+        cloudfoundry-community:
+          tenant: CF Community
+          rights:
+            - role: engineer
+
+  - name:       UAA
+  ...
+```
+
+## UAA
+
+This configuration gives you a single admin user with the password "PASSWORD"
+
+### Have a working UAA instance
+
+### Install cf-uaac Gem
+
+```shell
+gem install cf-uaac
+uaac version
+```
+
+### Authenticate to UAA
+
+```shell
+uaac target --skip-ssl-validation http(s)://<your uaa address>
+
+uaac token client get admin
+```
+
+### Create a new UAA Client for SHIELD
+
+```shell
+uaac client add shield-dev \
+  --name S.H.I.E.L.D. \
+  --scope openid \
+  --authorities uaa.none \
+  --authorized_grant_types authorization_code \
+  --redirect_uri http://<YOUR_SHIELD_ADDRESSS_HERE>/auth/oauth/<YOUR_UAA_IDENTIFIER_FROM_YOUR_SHIELD_CONFIG_HERE>/redir \
+  --access_token_validity  180 \
+  --refresh_token_validity 180 \
+  --secret s.h.i.e.l.d.
+```
+
+SHIELD requires the openid scope as this scope is required to access the /userinfo endpoint, which gives SHIELD the various standard user profile/group fields that it needs in order to exectue on the mapping in the configuration.
+
+### Add the configuration for your UAA OAuth client
+
+This will have identical required fields to the Github OAuth client defined above, with the exception of the backend being `uaa`, and the mapping section.  For UAA, SHIELD maps based on a users scim rights rather than the Github team they belong to.
```

Comment:
Explain the configuration again, since UAA users may not have reviewed the GH docs (nor should they have to)